benchmark/single-source/StringBreadcrumbs.swift

```diff
@@ -0,0 +1,494 @@
+//===--- StringBreadcrumbs.swift ------------------------------*- swift -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2018 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import TestsUtils
+
+public let StringBreadcrumbs: [BenchmarkInfo] = [
+  UTF16ToIndex(workload: longASCIIWorkload, count: 5_000).info,
+  UTF16ToIndex(workload: longMixedWorkload, count: 5_000).info,
+  IndexToUTF16(workload: longASCIIWorkload, count: 5_000).info,
+  IndexToUTF16(workload: longMixedWorkload, count: 5_000).info,
+
+  UTF16ToIndexRange(workload: longASCIIWorkload, count: 1_000).info,
+  UTF16ToIndexRange(workload: longMixedWorkload, count: 1_000).info,
+  IndexToUTF16Range(workload: longASCIIWorkload, count: 1_000).info,
+  IndexToUTF16Range(workload: longMixedWorkload, count: 1_000).info,
+
+  CopyUTF16CodeUnits(workload: shortASCIIWorkload, count: 500).info,
+  CopyUTF16CodeUnits(workload: shortMixedWorkload, count: 500).info,
+
+  MutatedUTF16ToIndex(workload: shortASCIIWorkload, count: 50).info,
+  MutatedUTF16ToIndex(workload: shortMixedWorkload, count: 50).info,
+  MutatedIndexToUTF16(workload: shortASCIIWorkload, count: 50).info,
+  MutatedIndexToUTF16(workload: shortMixedWorkload, count: 50).info,
+]
```

Comment:
And these names are over 40… :-/

Reply:
I don't see a way to shorten them without compromising clarity; they're already quite dense.

Reply:
I see two opportunities to shorten the final combined names:

1. `Breadcrumbs` as a family name is unique and memorable enough.
2. The `long` and `short` variants seem to be exclusively divided between the other 4 groups, so these prefixes can be dropped.

With these changes, the longest names would be  is `MutatedUTF16ToIndex`, the final results would be 😎 37 characters long: 

````
✅ Breadcrumbs.MutatedUTF16ToIndex.ASCII
✅ Breadcrumbs.MutatedIndexToUTF16.Mixed
````

Reply:
I'm okay with just Breadcrumbs. `longASCII` and `shortASCII` are quite different workloads though. 

I held my nose and replaced `Index` with `Idx`, `shortASCII` with `ASCII` and `shortMixed` with `Mixed`, leaving the `long`* variants intact. This leaves the names ugly but acceptable, and it gets them all below 40.

Reply:
> longASCII and shortASCII are quite different workloads though.
How so? I only see the same string repeated different times. That seems to me just like a different way to scale the workload size, instead of setting the inner-loop multiplier. Am I mistaken? Is the difference some kind of small string optimization behind the scenes? If so, this should probably be also documented in the comments.